### ISSUE TITLE
[TS] <DraggableCore>'s event handlers are not using React events, but the typing said so

### ISSUE THREAD

**Issue by be5invis** [react-grid-layout/react-draggable#373]
```ts
export type DraggableEventHandler = (e: React.MouseEvent<HTMLElement> | React.TouchEvent<HTMLElement>, data: DraggableData) => void | false;
```

should be 

```ts
export type DraggableEventHandler = (e: MouseEvent | TouchEvent, data: DraggableData) => void | false;
```

**Comment by be5invis** [comment]
@matoilic @STRML

**Comment by STRML** [comment]
What's the actual error?

**Comment by be5invis** [comment]
@be5invis 
In my app, from the typing info I used `e.presist()` and grabbed the native event via `e.nativeEvent` and it failed like `Uncaught TypeError: event.persist is not a function`.
Your TS typing in 3.1 is misguiding people.

**Comment by STRML** [comment]
It is a `React.MouseEvent` and `persist` is defined. What is the actual code you are running?

**Comment by be5invis** [comment]
I am using `DraggableCore`. Maybe DraggableCore is not using React events?

Some code:

```tsx
function handleDrag(e: React.MouseEvent<HTMLElement> | React.TouchEvent<HTMLElement>, d: DraggableData) {
	e.persist();
	somethingElse(e.nativeEvent);
}

<DraggableCore
	onDrag={handleDrag}
>
	<div>
		test
	</div>
</DraggableCore>
```

**Comment by STRML** [comment]
Since `SyntheticEvent` is pooled the performance impact will be negligible. We can't just adjust the typedefs - for instance, if the drag stop is initiated by a mouseup outside of the wrapped element, it will be a regular `MouseEvent`, not a `React.MouseEvent`. So where your mouse happens to be affects the callback's shape.

**Comment by STRML** [comment]
I don't see any good way to arbitrarily wrap a native event in a `SyntheticEvent`, so my proposal is to change the typedef:

```js
  export type DraggableEventHandler = (
    e: React.MouseEvent<HTMLElement> | React.TouchEvent<HTMLElement> | MouseEvent | TouchEvent,
    data: DraggableData
  ) => void | false;
```

Please try it out and open a PR if it works for you.

**Comment by be5invis** [comment]
@STRML LGTM
Perhaps this is more accurate
```ts
export type DraggableEventHandler = (
    e: React.MouseEvent<HTMLElement | SVGElement> | React.TouchEvent<HTMLElement | SVGElement> | MouseEvent | TouchEvent,
    data: DraggableData
  ) => void | false;
```

**Comment by STRML** [comment]
Fixed in #374
